### DISCUSSION TITLE
Progress tracking

### DISCUSSION THREAD

**Discussion by nikhilwoodruff** [PolicyEngine/policyengine.py#15] [General]
Centralised progress tracking for this project, with some overlap with our meeting notes [doc](https://docs.google.com/document/d/1bX6JQPHw-Bz13_n-SnJNE0qCXDr_KTe6cpaKvsv1BSc/edit).

**Comment by nikhilwoodruff** [comment]
## Weekly goals w/c 24th June

* Add basic Pytest test Inequality metrics asserting that we match the PolicyEngine numbers (passing)
* Add logic and tests (same as above) for all 6 poverty metrics

**Reply by nikhilwoodruff** [reply]
Complete! Ran into extra issues on the poverty metrics due to unavoidable error.

**Reply by masterismail** [reply]
## Weekly Learnings

This week, I learned about Pytest and GitHub workflows, as well as GitHub Actions.

**Comment by nikhilwoodruff** [comment]
# Weekly goals w/c 2nd July

* Finish six poverty metrics
* Add budgetary impact metrics
* Tests for those

**Reply by nikhilwoodruff** [reply]
All good here

**Reply by masterismail** [reply]
## Weekly Learnings

This week, I focused on improving my skills in two key areas:

### Pytest
- **Managing Multiple Files:** I learned how to effectively write and organize Pytest tests when dealing with multiple files.
- **Advanced Features:** I explored advanced Pytest features like parameterized tests, fixtures.

### Efficient File Management in Python Packages
- **Code Organization:** I discovered best practices for organizing code and resources within a Python package.
- **Dependency Management:** I gained insights into managing package dependencies and ensuring compatibility.

**Comment by nikhilwoodruff** [comment]
# Weekly goals w/c 8th July

* Finish vast majority of outstanding metrics
* Add tests for any metrics added

**Reply by nikhilwoodruff** [reply]
Complete! Just need a few more metrics

**Comment by nikhilwoodruff** [comment]
# Weekly goals w/c 15th July

* Finish remaining metrics in #11 
* Add tests and documentation for all remaining metrics
